saleor/static/dashboard-next/discounts/views/VoucherDetails.tsx

```diff
@@ -341,11 +346,15 @@ export const VoucherDetails: React.StatelessComponent<VoucherDetailsProps> = ({
                                         "SHIPPING"
                                           ? VoucherTypeEnum.SHIPPING
                                           : formData.type,
-                                      usageLimit: formData.usageLimit
+                                      usageLimit:
+                                        formData.hasUsageLimit &&
+                                        formData.usageLimit.length > 0
```

Comment:
This looks like `form.usageLimit` could be an array. It's even more weird because it's defined as `number`.
Let's type it as `string` as it really is `string`, then compare it here to `''`.